src/Psl/Xml/Error/XmlError.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Error;
+
+use LibXMLError;
+use Psl\Str;
+
+use const Psl\Xml\ERR_ERROR;
+use const Psl\Xml\ERR_FATAL;
+use const Psl\Xml\ERR_WARNING;
+
+final class XmlError
+{
+    private int $level;
+    private int $code;
+    private int $column;
+    private string $message;
+    private string $file;
+    private int $line;
```

Comment:
we should we make these public + add `@psalm-readonly` ? ( also, the class should marked immutable )

Reply:
If the end-user isn't using psalm, the properties can be changed.
So I'dd say it makes more sense to keep it as-is so that it cannot be abused by design.

(Otherwise we might just stick with regular LibXmlErrors - since they have public props)

WDYT?

Reply:
It's times like these I'm tempted to use https://github.com/Roave/you-are-using-it-wrong/ 😛 but want to keep it dependency-free. 

Anyway, let's stick with the getters for now.

Reply:
Than you can only use it with latest PHP version ... which doesnt bother me personal but wouldnt be optimal for a package like this :)